src/platform/silabs/wifi/wf200/WifiInterface.cpp

```diff
@@ -115,19 +115,17 @@ bool hasNotifiedWifiConnectivity = false;
 static uint8_t retryJoin         = 0;
 bool retryInProgress             = false;
 
-#ifdef SL_WFX_CONFIG_SCAN
 static struct scan_result_holder
 {
     struct scan_result_holder * next;
     wfx_wifi_scan_result scan;
 } * scan_save;
 static uint8_t scan_count = 0;
-static void (*scan_cb)(wfx_wifi_scan_result_t *); /* user-callback - when scan is done */
-static char * scan_ssid;                          /* Which one are we scanning for */
-size_t scan_ssid_length = 0;
+static ScanCallback scan_cb;              /* user-callback - when scan is done */
+static uint8_t * scan_ssid     = nullptr; /* Which one are we scanning for */
+static size_t scan_ssid_length = 0;
```

Comment:
I recommend you use some marker to identify the global variables. I usually prefix them with a single underscore, but a prefix would do as well.

Reply:
This will be addressed once we start the clean up of the WifiInterface implementations. The goal here is to fix the public APIs first.